auto_nag/scripts/triage_owner_rotations.py

```diff
@@ -0,0 +1,107 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+from typing import List, Set
+
+from libmozdata.bugzilla import BugzillaComponent
+from requests import HTTPError
+from tenacity import (
+    RetryError,
+    retry,
+    retry_if_exception_message,
+    stop_after_attempt,
+    wait_exponential,
+)
+
+from auto_nag import logger
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.component_triagers import ComponentName, ComponentTriagers, TriageOwner
+
+
+class TriageOwnerRotations(BzCleaner):
+    def __init__(
+        self,
+        excluded_components: List[str] = [],
+        excluded_teams: List[str] = [
+            "DOM LWS",
+            "Layout",
+        ],
+    ) -> None:
+        """Constructor
+
+        Args:
+            excluded_components: components to be excluded from the triage owner
+                rotation.
+            excluded_teams: teams to excluded all of their components when
+                performing the triage owner rotation.
+        """
+        super().__init__()
+        self.component_triagers = ComponentTriagers(excluded_components, excluded_teams)
+        self.query_url = None
+        self.has_put_errors = False
+
+    def description(self) -> str:
+        return "Triage owners that got updated"
+
+    def get_extra_for_template(self):
+        return {"has_put_errors": self.has_put_errors}
+
+    def _update_triage_owners(
+        self, new_triage_owners: List[TriageOwner]
+    ) -> Set[ComponentName]:
+        failures = set()
+        for new_triager in new_triage_owners:
+            if self.dryrun or self.test_mode:
+                logger.info(
+                    "The triage owner for '%s' will be: '%s'",
+                    new_triager.component,
+                    new_triager.bugzilla_email,
+                )
```

Comment:
Nit: maybe always log this, so remove the if and replace the else with `is not self.dryrun and not self.test_mode`

Reply:
As a side note: what is "test mode"? I had never heard of it :P

Reply:
> As a side note: what is "test mode"? I had never heard of it :P

Me too, I just matched `BzCleaner` behaviour; it seems a configuration that can be set in the `tools.json` file:

https://github.com/mozilla/relman-auto-nag/blob/179462fd873a93aaca104196479483f2e728279b/auto_nag/bzcleaner.py#L576

https://github.com/mozilla/relman-auto-nag/blob/aa8da65e7d6b4364c0994644180e9195daef0905/auto_nag/scripts/configs/tools.json#L43




> Nit: maybe always log this, so remove the if and replace the else with `is not self.dryrun and not self.test_mode`

Done in d3756d6184cbdcaeb785ca6bd16bfcaa72e1df2b